File: Bookify - Final Project/book-store/src/app/common/purchase.ts
import { Address } from './address';
import { Customer } from './customer';
import { Order } from './order';
import { OrderItem } from './order-item';

/**
 * Represents a purchase with customer, shipping and billing addresses, order, and order items.
 */

export class Purchase {
  customer!: Customer;
  shippingAddress!: Address;
  billingAddress!: Address;
  order!: Order;
  orderItems!: OrderItem[];
}

File: Bookify - Final Project/backend/bookstore-ecommerce/src/main/java/com/bookstore/bookstoreecommerce/dto/Purchase.java
package com.bookstore.bookstoreecommerce.dto;

import com.bookstore.bookstoreecommerce.entity.Address;
import com.bookstore.bookstoreecommerce.entity.Customer;
import com.bookstore.bookstoreecommerce.entity.Order;
import com.bookstore.bookstoreecommerce.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {

    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}
